File: algorithms/cses-problemset/introductory/mex_grid_construction.py
n = int(input())

mex_mat = [[0 for _ in range(n)] for _ in range(n)]


def mex(lst):
    return min(set(range(len(lst) + 1)) - set(lst))


# at worst case O(n^3 log n)
for i in range(n):
    for j in range(n):
        selected = [mex_mat[i][k] for k in range(j)] + [mex_mat[k][j] for k in range(i)]
        mex_mat[i][j] = mex(selected)

for i in range(n):
    for j in range(n):
        print(mex_mat[i][j], end=" ")
    print()

File: algorithms/cses-problemset/introductory/mex_grid_construction.go
//go:build ignore

package main

import (
	"fmt"
	"sort"
)

func mex(list []int) int {
	sort.Ints(list)
	for i := range len(list) {
		if list[i] != i {
			return i
		}
	}
	return len(list)
}

func main() {
	var n int
	fmt.Scanf("%d", &n)

	mexMat := make([][]int, n)

	for i := range n {
		mexMat[i] = make([]int, n)
		for j := range n {
			var selection []int
			for k := range j {
				selection = append(selection, mexMat[i][k])
			}
			for k := range i {
				selection = append(selection, mexMat[k][j])
			}

			mexMat[i][j] = mex(selection)
		}
	}

	for i := range n {
		for j := range n {
			print(mexMat[i][j], " ")
		}
		println()
	}
}
